lib/azure/armrest/wrapper/base_wrapper.rb

```diff
@@ -0,0 +1,81 @@
+require 'delegate'
+require 'ostruct'
+
+module Azure
+  module ArmRest
+    # Base class for JSON wrapper classes. Each Service class should have
+    # a corresponding class that wraps the JSON it collects, and each of
+    # them should subclass this base class.
+    class BaseWrapper < Delegator
```

Comment:
You want BaseModel or just Model? I guess I worry about name clashes, but it's not a huge deal.